docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,290 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+TypeScript!
+###########
+
+daml2ts : TypeScript of DAML
+============================
+
+Introduction
+------------
+
+``daml2ts`` generates TypeScript for interacting with DAML via the `json-api <../json-api/index.html>`_ .
+
+``daml2ts`` inputs are compiled DAML modules. ``daml2ts`` outputs are TypeScript source files containing definitions modeling the DAML entities found.
+
+The code ``daml2ts`` generates is in terms of the package `@daml/types <https://github.com/digital-asset/daml/tree/master/language-support/ts/daml-types>`_.
+
+Primitives: @daml/types
+-----------------------
+
+To understand the TypeScript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the primitives provided by @daml/types.
+
+**Interfaces**:
+
+- ``Serializable<T>``
+- ``Template<T extends object, K = unknown>``
+- ``Choice<T extends object, C, R, K = unknown>``
+
+  ..
+        An object that extends the ``Serializable<T>`` interface has a field ``decoder: () => jtv.Decoder<T>``.
+        An object that extends the ``Template<T, K>`` interface has fields ``templateId: string``, ``keyDecoder: ()=> jtv.Decoder<K>``, ``Archive: Choice<T, {}, {}>`` and extends ``Serializable<T>``.
+        An object that extends the ``Choice<T, C, R, K>`` interface has fields ``template: () => Template<T, K>``, ``argumentDecoder: () => jtv.Decoder<C>``, ``resultDecoder: () => jtv.Decoder<R>`` and ``choiceName: string``.
+
+**Types**:
+
++-------------------+------------------+----------------------------------+
+| TypeScript        | DAML equiv.      | TypeScript definition            |
```

Comment:
The first column should be the DAML type. That's what we get as input and users want to know how it gets converted into TS. Maybe this whole section should make this clear by having some like `Primitive DAML types` as its heading and then explaining that `@daml/types` contains TS equivalents of these types.

Reply:
I will alter the table to lead with the DAML type and change the heading.

Reply:
✔️